src/systems/systems.jl

```diff
@@ -49,6 +49,9 @@ function structural_simplify(
     if newsys isa ODESystem || has_parent(newsys)
         @set! newsys.parent = complete(sys; split, flatten = false)
     end
+    for pass in additional_passes 
+        newsys = pass(newsys)
+    end
```

Comment:
```suggestion
    for pass in additional_passes 
        newsys = pass(newsys)
    end
    if newsys isa ODESystem || has_parent(newsys)
        @set! newsys.parent = complete(sys; split, flatten = false)
    end
```

This way regardless of the passes, the system has the appropriate parent. Otherwise if the pass doesn't forward the parent to the transformed system, it'll break user workflows.